addons/web_editor/static/src/js/wysiwyg/MoveNodePlugin.js

```diff
@@ -0,0 +1,374 @@
+        }
+    }
+    _getMovableElements() {
+        return [...new Set([...this._editable.querySelectorAll(allowedElements)])]
+            .filter((node) =>
+                node.parentElement && node.parentElement.isContentEditable &&
+                isNodeMovable(node)
+            );
+    }
+    _getDroppableElements(draggableNode) {
+        return this._getMovableElements().filter((node) =>
+            !closestElement(node.parentElement, { predicate: (n) => n === draggableNode })
+        );
+    }
+    _setMovableElement(movableElement) {
+        this._removeMoveWidget && this._removeMoveWidget();
+        this._currentMovableElement = movableElement;
+        this._editor.disableAvatarForElement(movableElement);
+
+        const containerRect = this._widgetContainer.getBoundingClientRect();
+        const anchorBlockRect = this._currentMovableElement.getBoundingClientRect();
+        const closestList = closestElement(this._currentMovableElement, { selector: 'ul, ol' }); // Prevent overlap bullets.
+        const anchorX = closestList ? closestList.getBoundingClientRect().x : anchorBlockRect.x;
+        let anchorY = anchorBlockRect.y;
+        if (this._currentMovableElement.tagName.match(/H[1-6]/)) {
+            anchorY += (anchorBlockRect.height - widgetMoveSize) / 2;
+        }
+
+        this._moveWidget = this._document.createElement('div');
+        this._moveWidget.className = 'oe-sidewidget-move fa fa-sort';
+        this._widgetContainer.append(this._moveWidget);
+
+        let moveWidgetOffsetTop = 0;
+        if (movableElement.tagName === 'HR') {
+            const style = getComputedStyle(movableElement);
+            moveWidgetOffsetTop = parseInt(style.marginTop, 10) || 0;
+        }
+
+        this._moveWidget.style.width = `${widgetMoveSize}px`;
+        this._moveWidget.style.height = `${widgetMoveSize}px`;
+        this._moveWidget.style.top = `${anchorY - containerRect.y - moveWidgetOffsetTop}px`;
+        this._moveWidget.style.left = `${anchorX - containerRect.x - widgetContainerWidth}px`;
+        const timeout = setTimeout(()=> {
+            return this._moveWidget.classList.add('oe-sidewidget-move--show');
+        });
+
+        // Keep a reference for the closure.
+        const moveWidget = this._moveWidget;
+        this._removeMoveWidget = () => {
+            clearTimeout(timeout);
+            this._editor.enableAvatars(movableElement);
+            moveWidget.classList.remove('oe-sidewidget-move--show');
+            const transitionend = () => {
+                moveWidget.removeEventListener('transitionend', transitionend);
+                moveWidget.remove();
+            };
+            moveWidget.addEventListener('transitionend', transitionend);
+
+            // Ensure the widget is removed even if the the transition does not end.
+            setTimeout(() => moveWidget.remove(), 200);
+            this._removeMoveWidget = undefined;
+            this._currentMovableElement = undefined;
```

Comment:
Oh but you _do_ have this kind of variable _anyway_... In fact it looks like the only values that you're missing to have `_removeMoveWidget` be a proper method of the class is the avatars stuff...